Add BIND documentation (#52)
opnsense/docs: source/manual/how-tos/bind.rst

===========
BIND Plugin
===========

-------
History
-------

The history of the Bind plugin was a user request on OPNsense subreddit to create a 
plugin with a full-featured DNS server, also able to manage zonefiles with the most

> popular ressource records. In the beginning the plugin was build with only general 

Reviewer: ```suggestion
popular resource records. In the beginning the plugin was built with only general 
```

features so the community can contribute and adding wished features with a friendly

> review of the OPNsense Team.

Reviewer: ```suggestion
review of the OPNsense team.
```

via BIND Reverse Policy Zones.

For version 2.0 it is planned to offer full zone-file management.

------------
Installation
------------

First of all, go to **System->Firmware->Plugins** and install **os-bind**.
You will finde the plugin at **Services->BIND**.

----------------
General Settings
----------------

:Enable BIND Daemon:
    Enable the BIND service.
:Listen IPs:
    Set the IP addresses the daemon should listen on.
:Listen IPv6:
    Set the IPv6 addresses the daemon should listen on.
:Listen Port:
    Set the port the daemon should listen on. Per default the port is 53530 to not
    interfere with existing Unbound/dnsmasq setups. If you want to switch to BIND 
    only, make sure to stop Unbound and dnsmasq.
    
    
:DNS Forwarders:
    A list of IP addresses BIND will forward unknown dns request to. If empty BIND
    tries to resolve directly via the root servers.
:Logsize in MB:
    The amount for each logfile it can grow.
:Maximum Cache Size:
    This is the amount of RAM (in percent) the daemon can use for caching. 
:Recursion:
    You have to set a list of networks via **ACL** tab to allow them using recursion
    against BIND.
:DNSSec Validation:

>     Wheter to enable or disable DNSSec validation. 

Reviewer: ```suggestion
    Whether to enable or disable DNSSec validation. 
```

    
-----
DNSBL
-----

:Enable DNSBL:
    Enable the DNSBL service. BIND will be configured for Reverse Policy Zones to 
    blacklist domains. Choose below the lists to use for blacklist categories.
:Type of DNSBL:
    Here you can select the lists to use. Do not just select all and save. There are

>     websites don't loading content when nested Ad's are not loaded.

Reviewer: ```suggestion
    websites not loading content when nested ads are not loaded.
```

:Whitelist Domains:
    When a website is blocked due to a false positiv you can enter the domain name here

>     so it get's whitelisted before the blacklists come into play.

Reviewer: ```suggestion
    so it is whitelisted before the blacklists come into play.
```
